### PR TITLE
Allow CPU node types

### PR DESCRIPTION
**Pull request by zknill**

### CODE REVIEW

**Comment by noorvir** [comment]
Nice!

**Review by noorvir — APPROVED** [review]

File: api/types/hardware_test.go [hunk 1]
```diff
@@ -61,9 +65,11 @@ func TestSetDefaultValues(t *testing.T) {
...
 						Type: "test",
 					},
-					CPU: HardwareRequestRange{
-						Min: defaultMinCPU,
-						Max: 8,
+					CPU: CPU{
+						HardwareRequestRange: HardwareRequestRange{
+							Min: defaultMinCPU,
+							Max: 8,
+						},
```
Comment (mmacheerpuppy) [inline api/types/hardware_test.go:72]: why is the maximum 8, is this vcpu cores/threads?

File: api/types/spec.go [hunk 2]
```diff
@@ -106,9 +112,12 @@ func (m *NodeMetadataV1) GetHardwareSpec() HardwareSpec {
 				Max: m.GPUMemory,
 			},
 		},
-		CPU: HardwareRequestRange{
-			Min: m.VCPUs,
-			Max: m.VCPUs,
+		CPU: CPU{
+			Type: m.CpuType,
```
Comment (mmacheerpuppy) [inline api/types/spec.go:116]: where do we validate cputypes? does this happen elsewhere

File: api/types/hardware_test.go (hunk 1 above)
Comment (zknill) [reply api/types/hardware_test.go:72]: I actually don't know why we are using `8` here -- but this is a test, where I guess that's the test data we opted for. 
These sets of lines are just wrapping the existing `HardwareRequestRange` for cpu field in a dedicated object that also carries a `CPUType` field we can use.

File: providers/lambdalabs/node.go [hunk 3]
```diff
@@ -530,9 +530,11 @@ func getHardwareSpecFromInstance(instance client.Instance, gpuMem, gpuCount int)
 				Max: gpuMem,
 			},
 		},
-		CPU: types.HardwareRequestRange{
-			Min: instance.InstanceType.Specs.Vcpus,
-			Max: instance.InstanceType.Specs.Vcpus,
+		CPU: types.CPU{
+			HardwareRequestRange: types.HardwareRequestRange{
+				Min: instance.InstanceType.Specs.Vcpus,
```
[the remark this answers is not in the document]
Comment (mmacheerpuppy) [reply providers/lambdalabs/node.go:535]: oh i see

**Review by noorvir — APPROVED** [review]

**Pull request merged**